### ISSUE TITLE
[1.4] meteor build error using Joi above version 6.x.x

### ISSUE THREAD

**Issue by suhail-ansari-apconic** [meteor/meteor#7479]
Meteor build fails when using joi above version 6.x.x. 

You can easily recreate the problem by creating a new app and then add Joi above version 6. Then import Joi in one of your file. Try meteor build.

Error is as below:

Errors prevented bundling:                    
While minifying app code:                     

/home/sansari/.meteor/packages/standard-minifier-js/.1.1.8-rc.1.14v5ntt++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:196:18:
Unexpected token name «i», expected punc «;»
at new JS_Parse_Error
(/home/sansari/.meteor/packages/standard-minifier-js/.1.1.8-rc.1.14v5ntt++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:196:18)
at js_error
(/home/sansari/.meteor/packages/standard-minifier-js/.1.1.8-rc.1.14v5ntt++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:204:11)
at croak
(/home/sansari/.meteor/packages/standard-minifier-js/.1.1.8-rc.1.14v5ntt++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:675:9)
at token_error
(/home/sansari/.meteor/packages/standard-minifier-js/.1.1.8-rc.1.14v5ntt++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:683:9)
at expect_token
(/home/sansari/.meteor/packages/standard-minifier-js/.1.1.8-rc.1.14v5ntt++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:696:9)
at expect
(/home/sansari/.meteor/packages/standard-minifier-js/.1.1.8-rc.1.14v5ntt++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:699:36)
at regular_for
(/home/sansari/.meteor/packages/standard-minifier-js/.1.1.8-rc.1.14v5ntt++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:929:9)
at for_
(/home/sansari/.meteor/packages/standard-minifier-js/.1.1.8-rc.1.14v5ntt++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:925:16)
at
/home/sansari/.meteor/packages/standard-minifier-js/.1.1.8-rc.1.14v5ntt++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:807:24
at
/home/sansari/.meteor/packages/standard-minifier-js/.1.1.8-rc.1.14v5ntt++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:722:24

**Comment by benjamn** [comment]
Looking into this. Hopefully we'll be able to release a new version of `standard-minifier-js` and (thanks to core package version unpinning!) you'll be able to update without waiting for a new release of Meteor :)

**Comment by c9s** [comment]
Same here, but it's really hard to debug this issue.

I tried to dump the JS_Error structure, however there is no extra information for debugging 
```
?: Unexpected token: name (tmp) (line: 5127, col: 16, pos: 641044)
```

**Comment by c9s** [comment]
Here is the error stack:

```
   /Users/c9s/.meteor/packages/standard-minifier-js/.1.1.8.1tmo2du++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:197:18:
   Unexpected token: name (tmp)
   at new JS_Parse_Error
   (/Users/c9s/.meteor/packages/standard-minifier-js/.1.1.8.1tmo2du++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:197:18)
   at js_error
   (/Users/c9s/.meteor/packages/standard-minifier-js/.1.1.8.1tmo2du++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:205:11)
   at croak
   (/Users/c9s/.meteor/packages/standard-minifier-js/.1.1.8.1tmo2du++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:676:9)
   at token_error
   (/Users/c9s/.meteor/packages/standard-minifier-js/.1.1.8.1tmo2du++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:684:9)
   at unexpected
   (/Users/c9s/.meteor/packages/standard-minifier-js/.1.1.8.1tmo2du++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:690:9)
   at semicolon
   (/Users/c9s/.meteor/packages/standard-minifier-js/.1.1.8.1tmo2du++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:710:43)
   at simple_statement
   (/Users/c9s/.meteor/packages/standard-minifier-js/.1.1.8.1tmo2du++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:890:73)
   at
   /Users/c9s/.meteor/packages/standard-minifier-js/.1.1.8.1tmo2du++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:763:19
   at
   /Users/c9s/.meteor/packages/standard-minifier-js/.1.1.8.1tmo2du++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:723:24
   at block_
   (/Users/c9s/.meteor/packages/standard-minifier-js/.1.1.8.1tmo2du++os+web.browser+web.cordova/plugin.minifyStdJS.os/npm/node_modules/meteor/minifier-js/node_modules/uglify-js/lib/parse.js:1003:20)
```

**Comment by LeG3nDz** [comment]
Hi @benjamn 

This bug is fixed ? https://forums.meteor.com/t/meteor-1-4-build-error/27231

**Comment by abernix** [comment]
@LeG3nDz This is only "fixed" (simply by not minifying server files) in the `devel` branch right now.  If you are [running Meteor from a checkout](https://github.com/meteor/meteor/#slow-start-for-developers) you could test it right now.

It will likely be included in Meteor 1.4.1, possibly sooner, but no timeline for that right now.
